extensions/typescript-language-features/src/languageFeatures/diagnostics.ts

```diff
@@ -148,6 +151,76 @@ class DiagnosticSettings {
 	}
 }
 
+class DiagnosticsTelemetryManager extends Disposable {
+
+	private readonly _diagnosticCodesMap = new Map<number, number>();
+	private readonly _diagnosticSnapshotsMap = new Map<string, readonly vscode.Diagnostic[]>();
```

Comment:
You can use `ResourceMap` to store values for a given uri

Reply:
I noticed that the resource map takes as first argument a path normalizer. I am unsure if it is requires to normalize to a string in this case